src/averageAges.js

```diff
@@ -15,11 +16,15 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
```

Comment:
There is the same logic in all of this functions. So you can use the next function to calculate the average age for men, women, or children by passing in the appropriate filter and calculation functions as arguments:
function calculateAverageAge(people, filterFn, calculationFn) {
    const filteredPeople = people.filter(filterFn);
    const sum = filteredPeople.map(calculationFn).reduce((acc,val) => acc + val, 0);
    return sum / filteredPeople.length;
}

and other functions will be look like above

function calculateMenAverageAge(people, century) {
    const filterFn = century ? 
        (p) => p.sex === 'm' && Math.ceil(p.died / 100) === century : 
        (p) => p.sex === 'm';
    const calculationFn = (p) => p.died - p.born;
    return calculateAverageAge(people, filterFn, calculationFn);
}

function calculateWomenAverageAge(people, withChildren) {
    const filterFn = withChildren ? 
        (p) => p.sex === 'f' && people.find((p1) => p1.mother === p.name) : 
        (p) => p.sex === 'f';
    const calculationFn = (p) => p.died - p.born;
    return calculateAverageAge(people, filterFn, calculationFn);
}

function calculateAverageAgeDiff(people, onlyWithSon) {
    const filterFn = onlyWithSon ? 
        (p) => p.sex === 'm' && people.some((m) => m.name === p.mother) : 
        (p) => people.some((m) => m.name === p.mother);
    const calculationFn = (p) => p.born - people.find((m) => m.name === p.mother).born;
    return calculateAverageAge(people, filterFn, calculationFn);
}